### PR TITLE
Implemented Tools for deep linking

### PR DESCRIPTION
**Pull request by JosXa**
Allows
`url = helpers.create_deep_linked_url(bot.get_me().username, "Payload")`

### CODE REVIEW

**Comment by jh0ker** [comment]
Looks good, but the example seems very convoluted. Do we really need 3 levels of deep linking to demonstrate how a single function works?

**Comment by JosXa** [comment]
@jh0ker It's more about explaining what's possible with deep links and to spark the imagination a bit

**Comment by jh0ker** [comment]
Fair enough, but in that case perhaps you could make one of the levels use an inline button with a URL instead of just another text link

**Comment by JosXa** [comment]
Good point

File: telegram/utils/helpers.py [hunk 1]
```diff
@@ -139,3 +139,43 @@ def effective_message_type(entity):
...
+
+def create_deep_linked_url(bot_username, payload=None):
+    """
+    Creates a deep-linked URL for this ``bot_username`` with the specified ``payload``.
+    See  https://core.telegram.org/bots#deep-linking  to learn more.
+
+    The ``payload`` may consist of the following characters: ``A-Z, a-z, 0-9, _, -``
+
+    Note: Works well in conjunction with
+    ``CommandHandler("start", callback, Filters.regex('payload')``.
```
Comment (bomjacob) [inline telegram/utils/helpers.py]: Missing a paramthesis here?
Also shouldn't the filter be passed as a keyword argument?

File: telegram/utils/helpers.py [hunk 2]
```diff
@@ -139,3 +139,43 @@ def effective_message_type(entity):
...
+
+    Examples:
+        Example ``create_deep_linked_url(bot.get_me().username, "some-params")``
```
Comment (bomjacob) [inline telegram/utils/helpers.py]: "Examples: Example"
seems a bit redundant
(remove 2nd example text)

File: examples/deeplinking.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    """Reached through the CHECK_THIS_OUT payload"""
+    url = helpers.create_deep_linked_url(bot.get_me().username, SO_COOL)
+    text = "Awesome, you just accessed hidden functionality!\n\nContinue here: " + url
+    update.message.reply_text(text)
+
+
+def deep_linked_level_2(bot, update):
+    """Reached through the SO_COOL payload"""
+    url = helpers.create_deep_linked_url(bot.get_me().username, USING_ENTITIES)
+    text = "You can also mask the deep-linked URLs as links: " \
```
Comment (bomjacob) [inline examples/deeplinking.py]: Does this run on py2?

**Comment by tsnoam** [comment]
@JosXa 
What is the status of this PR? ( @bomjacob requested certain changes)

**Comment by JosXa** [comment]
Currently not having the time to fix it up. Anyone else interested in getting the kudos for this contribution? @bomjacob

**Comment by Bibo-Joshi** [comment]
Brought this PR up to date and implemented the requested changes.
Also made it possible create `startgroup` deep linking urls using @JosXa s new `helpers` method

**Pull request merged**